docs/guides/migration.md

```diff
@@ -0,0 +1,315 @@
+        "filename": "main.tf",
+        "start_line": 72
+    }
+    ```
+
+2. **Change the deprecated attributes:** Replace the deprecated attributes in your configuration file with the redefined attributes and execute the plan again.
+
+    ***Old Configuration***
+    ```hcl
+    resource "aci_bridge_domain" "terraform_bd" {
+        tenant_dn     = aci_tenant.terraform_tenant.id
+        name          = "terraform_bd"
+        ll_addr       = "::"
+        unk_mcast_act = "flood"
+    }
+    ```
+
+    ***New Configuration***
+    ```hcl
+    resource "aci_bridge_domain" "terraform_bd" {
+        parent_dn                     = aci_tenant.terraform_tenant.id
+        name                          = "terraform_bd"
+        link_local_ipv6_address       = "::"
+        l3_unknown_multicast_flooding = "flood"
+    }
+    ```
+
+    The `terraform plan` command should not display any warnings anymore.
+
+    ```bash
+    No changes. Your infrastructure matches the configuration.
+
+    Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.
+    ```
+
+### Step 5: Cleanup
+
+1. **Remove Backup (Optional):** If everything is working correctly, you can remove the backup state file.
+
+   ```bash
+   rm terraform.tfstate.backup
+   ```
+
+By following these steps, you can safely upgrade the ACI Terraform provider to a new version while ensuring that you have a backup of your current state in case anything goes wrong.
+
+## Optional Optimization Possibilites
+
+1. **Add relationships to parent resource:** Reduces the amount of resources in the configuration by including children inside the parent config. This will decrease the execution time by decreasing the size of the terraform graph and reducing the amount of REST API calls made towards APIC.
+
+    ***Old Configuration***
+    ```hcl
+    resource "aci_application_epg" "terraform_epg" {
+        parent_dn = aci_application_profile.terraform_ap.id
+        name      = "terraform_epg"
+    }
+
+    resource "aci_epg_to_domain" "terraform_epg_to_domain" {
+        application_epg_dn = aci_application_epg.terraform_epg.id
+        tdn                = "uni/phys-phys"
+    }
+    ```
+
+    ***New Configuration***
+    ```
+    resource "aci_application_epg" "terraform_epg" {
+        parent_dn = aci_application_profile.terraform_ap.id
+        name      = "terraform_epg"
+        relation_to_domains = [
+            {
+                target_dn = "uni/phys-phys"
+            }
+        ]
+    }
+    ```
+
+    !> Child resources should not be used in combination with nested attributes in parent resources. Doing so will result in unexpected behaviour.
+
+    In this scenario the child configuration should not be destroyed, but should be unmanaged (removed from state) before executing the plan. Terraform will display destroy intent in the `terraform plan` output when executed without unmanaging the child resource first.
```

Comment:
```suggestion
    In this scenario the child configuration should not be destroyed, but should be unmanaged (removed from state) before executing the plan. Terraform will display destroy intent in the `terraform plan` output when executed without unmanaging the child resource first.
```
What is the scenario? Maybe describe the scenario before showing the example or rephrase this. Also, add a sentence or link to the next part when you show how to unmanage.